lib/charms/postgresql_k8s/v0/postgresql_tls.py

```diff
@@ -130,11 +130,21 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             logger.error("An unknown certificate available.")
             return
 
-        self.charm.set_secret(
-            SCOPE, "chain", "\n".join(event.chain) if event.chain is not None else None
-        )
-        self.charm.set_secret(SCOPE, "cert", event.certificate)
-        self.charm.set_secret(SCOPE, "ca", event.ca)
+        if not event.certificate:
+            logger.debug("No certificate available.")
+            event.defer()
+            return
```

Comment:
In local deployments there were `None` certificates in the charm's internal secrets for the follower units. I think the `CertificateAvailableEvent` triggers for the application, not the unit.

Reply:
@dragomirp  Q: How MySQL charm handles this?